docs/generalutilities.md

```diff
@@ -0,0 +1,372 @@
+00:48:08.048 MSFT 63  3   4     40    c 3
+00:48:39.290 IBM  63  3   12    40    c 2
+00:57:47.067 AAPL 63  24  3     30    c 2
+01:08:00.945 AAPL 121 12  3     20    b 3
+
+```
+
+
+#### Example Unkeyed Table:
+We have the following table:
+```
+q)table
+time         sym  ask bid asize bsize a id
+------------------------------------------
+00:00:00.000 AAPL 73  90  3     30    a 1
+00:00:00.000 AMD  123 210 67    30    b 4
+00:00:00.000 IBM  34  64  87    43    b 4
+00:00:00.000 MSFT 100 200 43    40    c 1
+00:00:38.184 AMD  121 12  3     30    a 1
+00:01:25.332 AMD  121 3   3     30    b 4
+00:09:37.574 AAPL 63  32  3     30    b 3
+00:21:24.796 AAPL 63      12            4
+00:38:39.521 AAPL 121 3   5           a 3
+00:40:41.670 MSFT 63  3         20      4
+00:48:08.048 MSFT 63      4     40    c 3
+00:48:39.290 IBM  63  3   12    40      2
+00:57:47.067 AAPL     24  3     30      2
+..
+```
+Create the dictionary:
+```
+q)args:(`t`k)!(table;`sym)
+q)args
+t| `s#+`time`sym`ask`bid`asize`bsize`a`id!(`p#00:00:00.482 00:00:00.803 00:00..
+k| `sym
+
+```
+Pass the dictionary to the function:
+```
+q)ffill[args]
+time         sym  ask bid asize bsize a id
+------------------------------------------
+00:00:00.000 AAPL 73  90  3     30    a 1
+00:00:00.000 AMD  123 210 67    30    b 4
+00:00:00.000 IBM  34  64  87    43    b 4
+00:00:00.000 MSFT 100 200 43    40    c 1
+00:00:38.184 AMD  121 12  3     30    a 1
+00:01:25.332 AMD  121 3   3     30    b 4
+00:09:37.574 AAPL 63  32  3     30    b 3
+00:21:24.796 AAPL 63  32  12    30    b 4
+00:38:39.521 AAPL 121 3   5     30    a 3
+00:40:41.670 MSFT 63  3   43    20    c 4
+00:48:08.048 MSFT 63  3   4     40    c 3
+00:48:39.290 IBM  63  3   12    40    b 2
+00:57:47.067 AAPL 121 24  3     30    a 2
+..
+```
+
+#### Example Keyed Table:
+We have the following table:
+```
+q)ktable
+time        | sym  ask bid asize bsize a
+------------| --------------------------
+00:00:00.000| AAPL 73  90  3     30    a
+00:00:00.000| MSFT 100 200 43    40    c
+00:00:00.000| AMD  123 210 67    30    b
+00:00:00.000| IBM  34  64  87    43    b
+00:00:22.178| IBM  4   65        30
+00:01:20.951| AAPL 35  130 3           a
+00:06:06.973| MSFT 93  144 3     20
+00:08:15.333| AAPL 98  167 3     20
+00:08:23.291| IBM  69  84  12    30    b
+00:10:27.142| IBM  92  85              c
+..
+```
+Create the dictionary:
+```
+q)dict:(`t`k)!(ktable;`sym)
+```
+Pass the dictionary to the function:
+```
+q)ffill[dict]
+time        | sym  ask bid asize bsize a
+------------| --------------------------
+00:00:00.000| AAPL 73  90  3     30    a
+00:00:00.000| MSFT 100 200 43    40    c
+00:00:00.000| AMD  123 210 67    30    b
+00:00:00.000| IBM  34  64  87    43    b
+00:00:22.178| IBM  4   65  87    30    b
+00:01:20.951| AAPL 35  130 3     30    a
+00:06:06.973| MSFT 93  144 3     20    c
+00:08:15.333| AAPL 98  167 3     20    a
+00:08:23.291| IBM  69  84  12    30    b
+00:10:27.142| IBM  92  85  12    30    c
+..
```

Comment:
I'm not sure why there are so many examples.  keyed table and unkeyed should be the same?  I don't think one illustrates it better than the other.  Just do one example, and included a string column in it to be forward filled.